internal/hmyapi/apiv1/blockchain.go

```diff
@@ -685,10 +740,9 @@ func (s *PublicBlockChainAPI) getAllValidatorInformation(
 	}
 	for i := start; i < start+validatorsNum; i++ {
 		information, err := s.b.GetValidatorInformation(addresses[i], block)
-		if err != nil {
-			return nil, err
+		if err == nil {
+			validators[i-start] = information
```

Comment:
instead of making a fixed length slice and leaving some spots empty if validator not found for particular block, make the slice literal and append to it.